ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,451 @@
+        ('std_msgs', 'msg', 'Bool'))
+    :rtype: tuple
+    """
+    if '/' not in resource:
+        return '', '', resource
+    values = resource.split('/')
+    if len(values) != 3:
+        raise ValueError('Resource name "{}" is malformed'.format(resource))
+    return tuple(values)
+
+
+def get_templates_dir():
+    """
+    Return template directory.
+
+    :returns: the directory of the template directory
+    :rtype: str
+    """
+    return os.path.join(os.path.dirname(__file__), _TEMPLATES_DIR)
+
+
+def copy_css_style(folder_name):
+    """
+    Copy the css style files in the folder_name.
+
+    :param folder_name: name of the folder where the style css files will be copied
+    :type folder_name: str
+    """
+    for style in ['styles.css', 'msg-styles.css']:
+        shutil.copy(os.path.join(get_templates_dir(), style), os.path.join(folder_name, style))
+
+
+def load_template(input_filename):
+    """
+    Look up file within rosdoc ROS package.
+
+    return its content, may sys.exit on error.
+    Contents are cached with filename as key.
+
+    :param input_filename: name of the template to be loaded
+    :type input_filename: str
+    :returns: cached file contents and filename of the template concat with
+        the template directory
+    :rtype: tuple
+    """
+    filename = os.path.join(get_templates_dir(), input_filename)
+    if not os.path.isfile(filename):
+        raise FileNotFoundError(
+            errno.ENOENT, os.strerror(errno.ENOENT), filename)
+    with open(filename, 'r') as f:
+        content = f.read()
+        if not content:
+            raise Exception("Template file '%s' is empty\n" % (filename))
+        return content, filename
+
+
+def generate_interface_documentation(interface, interface_template, file_output_path,
+                                     documentation_data, generate_text_from_spec):
+    """
+    Generate documentation for a single ROSIDL interface.
+
+    This function write in a file the message static HTML site.
+
+    :param interface: name of the interface
+    :type interface: str
+    :param interface_template: name of the template
+    :type interface_template: str
+    :param file_output_path: name of the file where the template will be written
+        once filled
+    :type file_output_path: str
+    :param documentation_data: dictionary with the data to field the index template
+    :type documentation_data: dict
+    :param generate_text_from_spec: function with the logic to fill the compact
+        definition for a specific type of interface
+    :type generate_text_from_spec: function
+    """
+    package, interface_type, base_type = resource_name(interface)
+    file_path = get_interface_path(interface)
+    with open(file_path, 'r', encoding='utf-8') as h:
+        spec = h.read().rstrip()
+
+    compact_definition = generate_text_from_spec(package, base_type, spec)
+    documentation_data['raw_text'] = generate_raw_text(spec)
```

Comment:
Why should any of this logic happen in this Python module rather than in the template?

Reply:
moved to the template https://github.com/ros2/ros2_generate_interface_docs/pull/4/commits/f1b227ca70c5644da97c06ee3dfaf821832f8825

Reply:
Added local variable https://github.com/ros2/ros2_generate_interface_docs/pull/4/commits/484061f32c643b184566050e0214034eb70fa28e